### PR TITLE
[MAINTAIN-177] - Clear searched keywords by Clear filters button

### PR DESCRIPTION
**Pull request by manachynskyi**
This pr is going to add new `Clear keywords` button.

Steps for review:
- [ ] Visit activity finder 4 page
- [ ] go through all steps to the result page and perform a search by some keyword that won't return results (e.g. test) 
- [x] make sure there is a `Clear keyword` button present https://i.imgur.com/bbTDT8J.png (it should clear keywords)
- [ ] type keyword again (without results) and additionally select some filter
- [x] `Clear filters` buttons should appear. https://i.imgur.com/WBE4G9L.png
- [ ] Additionally, select several of any other filters
- [x] make sure you see buttons, each of them should leave **ONLY** a selected filter specified in the button title, while keywords shouldn't be cleared https://i.imgur.com/q2VAMFF.png

### CODE REVIEW

**Review by hamrant — APPROVED** [review]

**Review by duozersk — CHANGES_REQUESTED** [review]
I don't have access to the https://openy.atlassian.net/browse/MAINTAIN-177... the access was requested (Julia Z have asked Sarah B to provide access for me)
The "Clear filters" functionality was specifically designed to not clear the keywords filter - so we need to check with initial stakeholders if this is fine to change it globally (for all the "Clear filters" appearances in the application).

**Comment by podarok** [comment]
@sarah-halby we should check with Ben/Craig if keywords search should be cleared when using "Clear filters" button
I second @duozersk - I remember we initially disabled clearing keywords. https://openy.atlassian.net/browse/MAINTAIN-177?focusedCommentId=12607

**Comment by podarok** [comment]
As a suggestion - we should make this feature optional and disabled by default. When someone needs it enabled - should be a checkbox on the Paragraph configuration level
@manachynskyi @duozersk

**Review by podarok — APPROVED** [review]
Code wise looks good

File: openy_af4_vue_app/src/App.vue [hunk 1]
```diff
@@ -806,6 +808,9 @@ export default {
           : this.defaults[key]
       }
     },
+    clearKeywords() {
+      this['searchKeywords'] = ''
```
Comment (duozersk) [inline openy_af4_vue_app/src/App.vue]: Strange notation... why not `this.searchKeywords` like in all the other places?

File: openy_af4_vue_app/src/components/NoResults.vue [hunk 2]
```diff
@@ -100,11 +115,21 @@ export default {
...
+    selectedDimensionsAndKeywords() {
+      return this.selectedDimensions + (this.searchKeywords.length ? 1 : 0)
     }
   },
   methods: {
     onChoice(choice) {
+      // If we leave some specific filter, we should clear all others including the keywords.
+      if (choice) {
+        this.clearKeywords()
+      }
```
Comment (duozersk) [inline openy_af4_vue_app/src/components/NoResults.vue]: No, it is definitely a change to the logic we had before. We need to clear keywords only when the user explicitly wants to clear them.
And even if we need to clear them - this action should not be in this component, this logic should be in the App.vue component level, not here.

I would just remove this piece for now.

**Review by duozersk — CHANGES_REQUESTED** [review]

File: openy_af4_vue_app/src/components/NoResults.vue (hunk 2 above)
Comment (manachynskyi) [reply openy_af4_vue_app/src/components/NoResults.vue]: @duozersk This piece will be invoked only when criteria are displayed [image](https://i.imgur.com/TaFF7gP.png ). If we remove this piece, the logic will little bit strange, the user chose the most important criteria, but not all filters will be cleared.

The logic of clearing located in App.vue, in this pieace we just emit event.

**Comment by podarok** [comment]
this is approved by Ben, we can move forward with merging
Let me know when this is ready for final review @duozersk @manachynskyi

File: openy_af4_vue_app/src/components/NoResults.vue (hunk 2 above)
Comment (duozersk) [reply openy_af4_vue_app/src/components/NoResults.vue]: @manachynskyi ok, let's have a call then; I'll find you ;)

File: openy_af4_vue_app/src/App.vue (hunk 1 above)
Comment (manachynskyi) [reply openy_af4_vue_app/src/App.vue]: Fixed.

File: openy_af4_vue_app/src/components/NoResults.vue (hunk 2 above)
Comment (manachynskyi) [reply openy_af4_vue_app/src/components/NoResults.vue]: Fixed.

**Review by duozersk — APPROVED** [review]

**Review by podarok — APPROVED** [review]
Code looks good
Merging for dev testing by QA on sandboxes

**Pull request merged**